CONTRIBUTING.rst

```diff
@@ -0,0 +1,153 @@
+  own code.
+- Describe what actually happened. Include the full traceback if there was an
+  exception.
+- List your Python, Flask, and Werkzeug versions. If possible, check if this
+  issue is already fixed in the repository.
+
+.. _minimal, complete, and verifiable example: https://stackoverflow.com/help/mcve
+
+Submitting patches
+------------------
+
+- Include tests if your patch is supposed to solve a bug, and explain
+  clearly under which circumstances the bug happens. Make sure the test fails
+  without your patch.
+- Try to follow `PEP8`_, but you may ignore the line length limit if following
+  it would make the code uglier.
+
+First time setup
+~~~~~~~~~~~~~~~~
+
+- Download and install the `latest version of git`_.
+- Configure git with your `username`_ and `email`_::
+
+        git config --global user.name 'your name'
+        git config --global user.email 'your email'
+
+- Make sure you have a `GitHub account`_.
+- Fork Flask-SQLAlchemy to your GitHub account by clicking the `Fork`_ button.
+- `Clone`_ your GitHub fork locally::
+
+        git clone https://github.com/{username}/flask-sqlalchemy
+        cd flask-sqlalchemy
+
+- Add the main repository as a remote to update later::
+
+        git remote add pallets https://github.com/pallets/flask-sqlalchemy
+        git fetch pallets
+
+- Create a virtualenv::
+
+        python3 -m venv env
+        . env/bin/activate
+        # or "env\Scripts\activate" on Windows
+
+- Install Flask-SQLAlchemy in editable mode with development dependencies::
+
+        pip install -e ".[dev]"
+
+.. _GitHub account: https://github.com/join
+.. _latest version of git: https://git-scm.com/downloads
+.. _username: https://help.github.com/articles/setting-your-username-in-git/
+.. _email: https://help.github.com/articles/setting-your-email-in-git/
+.. _Fork: https://github.com/pallets/flask-sqlalchemy/fork
+.. _Clone: https://help.github.com/articles/fork-a-repo/#step-2-create-a-local-clone-of-your-fork
+
+Start coding
+~~~~~~~~~~~~
+
+- Create a branch to identify the issue you would like to work on (e.g.
+  ``2287-dry-test-suite``)
+- Using your favorite editor, make your changes, `committing as you go`_.
+- Try to follow `PEP8`_, but you may ignore the line length limit if following
+  it would make the code uglier.
+- Include tests that cover any code changes you make. Make sure the test fails
+  without your patch. `Run the tests. <contributing-testsuite_>`_.
+- Push your commits to GitHub and `create a pull request`_.
+- Celebrate 🎉
+
+.. _committing as you go: https://dont-be-afraid-to-commit.readthedocs.io/en/latest/git/commandlinegit.html#commit-your-changes
+.. _PEP8: https://pep8.org/
+.. _create a pull request: https://help.github.com/articles/creating-a-pull-request/
+
+.. _contributing-testsuite:
+
+Running the tests
+~~~~~~~~~~~~~~~~~
+
+Install the testing requirements::
+
+    pip install pytest mock tox coverage
```

Comment:
`pip install -e ".[tests]"`?

Reply:
See above - I don't think there's a `tests` extra in `setup.py` for this project.